translate/src/test_util.rs

```diff
@@ -36,35 +36,39 @@ pub struct MockTool {
 impl MockTool {
     /// Creates a new MockTool. The MockTool is boxed, because most users of MockTool will use it
     /// to create a `Box<dyn Tool>`.
-    pub fn new() -> Box<MockTool> {
+    pub fn new() -> MockTool {
         MockTool {
             name: "MockTool",
             might_write: Box::new(|_| MightWriteOutcome::Runnable([].into())),
             run: Box::new(|_| Ok(())),
         }
-        .into()
+    }
+
+    /// Returns this MockTool in a box. For use when a `Box<dyn Tool>` is needed.
```

Comment:
This is a nice convenience, but you don't actually _need_ this. You can just wrap the result of `.run()` in a `Box::new()` when needed.